lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -2935,7 +2942,7 @@ Type GenericSignatureBuilder::PotentialArchetype::getDependentType(
 
     // If we've resolved to an associated type, use it.
     if (auto assocType = getResolvedType())
-      return DependentMemberType::get(parentType, assocType);
+      return getResolvedDependentMemberType(parentType);
 
     return DependentMemberType::get(parentType, getNestedName());
```

Comment:
Oh, I forgot to include this, but this line is dead code. Why? Because potential archetypes with parents always have an associated type, therefore `getResolvedType()` always returns a non-null pointer.

Reply:
Ahh, this is a vestige from a time that was not true, when we still had potential archetypes that were as-yet unresolved to an associated type.